Topics/System_Design/DesignPatterns/Structural_Design_Patterns.md

```diff
@@ -0,0 +1,186 @@
+
+### Example Scenario: <a name="exampleb"></a>
+Suppose we have a class called "*EmployeeAtSchool*" which is the parent of  4 children classes called "*JanitorFullTime*",  "*JanitorPartTime*",  "*TeacherFullTime*" and "*TeacherPartTime*" respectively.  Let's visualize this clearly:
+```mermaid
+graph TD;
+A[EmployeeAtSchool]--Parent of--> B[JanitorFullTime]
+A[EmployeeAtSchool]--Parent of--> C[JanitorPartTime]
+A[EmployeeAtSchool]--Parent of--> D[TeacherFullTime]
+A[EmployeeAtSchool]--Parent of--> F[TeacherPartTime]
+linkStyle default stroke-width:3x,fill:none,stroke:purple;
+```
+Suppose we want to add another child class to "*EmployeeAtSchool*" class called "*Student*".  However, we cannot simply add the class "*Student*".  Based on the current structure and design of "*EmployeeAtSchool*" class,  we must add two child classes:  one for full-time student and one for part-time student. In other words, we must add the child classes "*StudentFullTime*" and "*StudentPartTime*" respectively to the "*EmployeeAtSchool*" class. We can already see how complex and badly designed and structured the "*EmployeeAtSchool*" class is becoming. 
+
+To solve this problem, we will use the Bridge Design Pattern. By its definition, in this scenario, we will create two classes. One of them is called  "*Employee*"  which cares about the types of employee only, and  the other is called  "*EmployeeStatus*" which only cares about the full-time and part-time types for employees. To connect these two classes, we will make sure that the "*Employee*" class **contains** an the object of type "*EmployeeStatus*". Let's demonstrate this:
+```mermaid
+graph TD;
+A[Employee]--Parent of--> B[Janitor]
+A[Employee]--Parent of--> C[Teacher]
+A[Employee]--Parent of--> D[Student]
+K[EmployeeStatus]--Parent of--> F[PartTime]
+K[EmployeeStatus]--Parent of--> M[FullTime]
+A-. Contains .-> K
+linkStyle 5 stroke-width:3x,fill:none,stroke:red;
+linkStyle default stroke-width:3x,fill:none,stroke:purple;
+```
+
+### Pros and Cons:  <a name="PCb"></a>
+
+ - Pros: Allows the client code to work with abstractions which is good for the user because he does not have to know about the implementation process. It also allows abstractions and implementations to be done for each class separately. 
+ - Cons: Causes the creation of more classes which might add complexity.
+ 
+## *Conclusion* <a name="conclu"></a>
+
+In summary, in this tutorial, we covered what **Structural Design Patterns** are, what are their characteristics and how and when they are useful. We also went through the **Adapter Design Pattern** and the **Bridge Design Pattern** respectively, which are two of the important design patterns in the category of Structural Design Patterns. To learn more about this categoty of design patterns visit [Structural Design Patterns]([Structural Design Patterns (refactoring.guru)](https://refactoring.guru/design-patterns/structural-patterns). 
```

Comment:
You duplicated the link text, so it appears oddly in the text.

Reply:
Thanks for the catch. Its a typo